### ISSUE TITLE
Potential problem in look_at()

### ISSUE THREAD

**Issue by JohnGrime** [godotengine/godot#49839]
### Godot version

3.3.2.stable.official

### System information

macOS 11.3.1 GLES3

### Issue description

When using the `look_at()` function in a script attached to a camera, there seems to be some sort of discretization effect occurring in my tests; for a camera positioned at e.g. `(0,90,0)` looking towards `(0,0,0)`, I see rendering artifacts. These artifacts persist for a small region around `(0,0,0)` on the `x,z` plane (approx. +/- 0.1 units).

This effect is particularly jarring when it manifests as part of an animated movement, producing an effect almost like a magnet snapping the camera view direction onto the target location at `(0,0,0)`.

I've attached an example project to demonstrate this effect; a camera is located at `(0,90,0)` above a simple plane, and the `look_at()` method is invoked in the camera's `_process()` method. The target position is slowly moved left and right via a `sin()` function, and the effect is visible when the camera looks in the vicinity of `(0,0,0)`.

Am I missing something important, here?

### Steps to reproduce

See issue description, and attached project file.

### Minimal reproduction project

[Weird.zip](https://github.com/godotengine/godot/files/6697007/Weird.zip)

**Comment by kleonc** [comment]
Looks more like a problem with `OmniLight`'s shadow. Duplicate of #25750?

- Plane's depth/width subdivision = 0:
![Godot_v3 3 2-stable_win64_sgiu2cLQ8G](https://user-images.githubusercontent.com/9283098/122996666-6c03f880-d3ab-11eb-96bb-c74e6370a841.png)

- Plane's depth/width subdivision = 3:
![Godot_v3 3 2-stable_win64_nV6YpehB9n](https://user-images.githubusercontent.com/9283098/122998137-fa2cae80-d3ac-11eb-9b4f-908c5732ce47.png)

**Comment by JohnGrime** [comment]
It's not just a rendering artifact, I think there's also a "jump" or "snap" effect on the camera view vector. At least there is for me! :)

**Comment by Calinou** [comment]
I can reproduce this on Linux, so this is not an OS-specific issue.

**Comment by kleonc** [comment]
> It's not just a rendering artifact, I think there's also a "jump" or "snap" effect on the camera view vector. At least there is for me! :)

You're right, the rendering artifact is gone when shadow is disabled in `OmniLight` but for `targetPos.x == targetPos.y == 0.0` and `abs(targetPos.z) < 0.403` it looks (incorrectly) like we're looking at `targetPos == (0, 0, 0)`.

**Comment by JohnGrime** [comment]
Simply moving the camera from `(0,90,0)` to `(1,90,0)` (but keeping the target as `(0,0,f(sin t))`) seems to fix the "snapping magnet" issue.

If I then change the dynamic camera target from `(0,0,f(sin t))` to `(1,0,f(sin t))` (i.e., the camera view vector is back along `(-1,0,0)`) someone puts the pesky magnet back onto the camera ...

**Comment by JohnGrime** [comment]
Smells like a numerical issue when calculating the inverse of a transform matrix, maybe?

Or perhaps a pesky `if (x < epilson) { y = constant; } /* prevent DBZ */` sort of issue somewhere?
